db/census_participant.go

```diff
@@ -479,6 +479,47 @@ func (ms *MongoStorage) SetBulkCensusParticipant(
 	return progressChan, nil
 }
 
+func (ms *MongoStorage) setBulkCensusParticipant(
+	ctx context.Context, censusID string, memberIDs []primitive.ObjectID,
+) error {
+	currentTime := time.Now()
+	docs := make([]mongo.WriteModel, 0, len(memberIDs))
+	for _, pid := range memberIDs {
+		// Create participant filter and document
+		id := pid.Hex()
+		censusParticipantsFilter := bson.M{
+			"participantID": id,
+			"censusId":      censusID,
+		}
+		participantDoc := &CensusParticipant{
+			ParticipantID: id,
+			CensusID:      censusID,
+			CreatedAt:     currentTime,
+			UpdatedAt:     currentTime,
+		}
+
+		// Create participant update document
+		updateParticipantDoc, err := dynamicUpdateDocument(participantDoc, nil)
+		if err != nil {
+			log.Warnw("failed to create update document for participant",
+				"error", err, "participantID", id)
+			continue
+		}
+
+		// Create participant upsert model
+		upsertCensusParticipantsModel := mongo.NewUpdateOneModel().
+			SetFilter(censusParticipantsFilter).
+			SetUpdate(updateParticipantDoc).
+			SetUpsert(true)
+		docs = append(docs, upsertCensusParticipantsModel)
+	}
+	// Unordered makes it continue on errors (e.g., one dup), but you can set Ordered=true if you prefer.
```

Comment:
"but you can set Ordered=true if you prefer"??

`// Unordered makes it continue on errors (e.g., one dup)` sounds enough